src/MassTransit/Logging/EnabledScope.cs

```diff
@@ -0,0 +1,36 @@
+namespace MassTransit.Logging
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.Extensions.Logging;
+
+
+    public readonly struct EnabledScope
+    {
+        readonly IDisposable _scope;
+        readonly IDictionary<string, object> _state;
+
+        public EnabledScope(ILogger logger)
+        {
+            _state = new Dictionary<string, object>();
+
+            _scope = logger.BeginScope(_state);
+        }
+
+        public void Add(string key, object value)
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            if (value == default)
```

Comment:
One thing I would like to notice, if you, for example, pass default for pass primitive types (`0` for int, or `false` for bool) it will try to remove it from the dictionary

Reply:
yeah, that was a choice to avoid all the empty tags in the logs.